### PR TITLE
feat(AI Transform Node): Reduce payload size

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary

if request returned 'maximum context length' error reduce payload size and try again

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-1866/improve-error-handling-request-entity-too-large-error

### CODE REVIEW

**Comment by michael-radency** [comment]
thanks for review @tomi, updated as suggested

**Review by tomi — COMMENTED** [review]
Thank you for splitting up the function. It's much easier to follow now. Couple comments more

File: packages/editor-ui/src/components/ButtonParameter/utils.ts [hunk 1]
```diff
@@ -57,6 +57,128 @@ export function getSchemas() {
...
 }
 
+//------ Reduce payload ------
+
+const calculateTokens = (item: object, averageTokenLength: number): number => {
+	return Math.ceil(JSON.stringify(item).length / averageTokenLength);
+};
+
+const calculateRemainingTokens = (error: Error) => {
+	const tokens = error.message.match(/\d+/g);
```
[the remark this answers is not in the document]
Comment (michael-radency) [reply packages/editor-ui/src/components/ButtonParameter/utils.ts:73]: updated as suggested, regarding other messages - we are checking message to  contain specific subsisting 'maximum context length'

**Review by tomi — APPROVED** [review]
🚀

**Comment by michael-radency** [comment]
thanks again @tomi , updated as suggested

**Comment by janober** [comment]
Got released with [`n8n@1.72.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.72.0)

**Pull request merged**